Client/Frontend/Browser/URIFixup.swift

```diff
@@ -11,7 +11,10 @@ class URIFixup {
 
         // First check if the URL includes a scheme. This will handle
         // all valid requests starting with "http://", "about:", etc.
-        if !(url?.scheme.isEmpty ?? true) {
+        // However, we ensure that the scheme is one that is listed in
+        // the official URI scheme list, so that other such search phrases
+        // like "filetype:" are recognised as searches rather than URLs.
+        if !(url?.scheme.isEmpty ?? true) && url!.schemeIsValid {
```

Comment:
The `isEmpty` check isn't needed anymore since your `schemeIsValid` will handle the empty case now.

How about we drop the `var url` above, then declare it locally? That would let us drop these ugly optional decorators:
`if let url = NSURL(string: trimmed) where url.schemeIsValid {`